crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -393,6 +396,16 @@ async fn app_validation_workflow_inner(
         }
     }
 
+    // "self-publish" warrants
```

Comment:
Are warrants supposed to be published like chain ops? If that is so, this doesn't trigger the publish workflow.

Reply:
They are supposed to, yes. When I say "self-publish" I mean "'publish' to myself" -- get them into my DHT table as if they were published to me.

As for publishing to others, that always happens with the authored table as the source.

Reply:
> As for publishing to others, that always happens with the authored table as the source.

Where does that happen?

Reply:
https://github.com/holochain/holochain/blob/6bc52eb593822fa4ec7f10026550f9f895e135cd/crates/holochain/src/core/workflow/app_validation_workflow.rs#L196-L198

Reply:
But where is sys validation or app validation triggered?

Reply:
Of the warrants themselves? When they are processed as incoming DhtOps https://github.com/holochain/holochain/blob/fb45cd3653ec2b36e2798ea2216988c4b5202487/crates/holochain/src/core/workflow/incoming_dht_ops_workflow.rs#L194-L198

Reply:
That will only happen when some other incoming ops after the warrants trigger this, right? It's not explicitly triggered for warrants AFAICT.

Reply:
`WarrantOp`s are `DhtOp`s, so they are "incoming" in exactly the same sense as `ChainOp`s (which were previously named "`DhtOp`")

Reply:
Yes, what I'm referring to is, if after creating a warrant it is supposed to be published, that it is not happening until some unrelated incoming op triggers the workflows.

Reply:
Hmm, you were asking about where sys validation gets triggered, which I showed you above. Earlier you asked about where publishing gets triggered (after creating a warrant), which I also showed, and which I think answers your most recent question: https://github.com/holochain/holochain/pull/4033#discussion_r1662976374

Is there a spot I'm missing?

Reply:
No, my question's answered. Warrants are created in app validation and publish is triggered from there.

Reply:
Yes. Warrants are also created in sys validation, and publish is also triggered from there